Quick_Deploy/vLLM/model_repository/vllm/1/model.py

```diff
@@ -0,0 +1,330 @@
+            max_num_seqs=vllm_engine_config.max_num_seqs,
+        )
+        self.llm = AsyncLLMEngine.from_engine_args(engine_args)
+
+        self._loop = asyncio.get_event_loop()
+        self._loop_thread = threading.Thread(
+            target=self.engine_loop, args=(self._loop,)
+        )
+        self._shutdown_event = asyncio.Event()
+        self._loop_thread.start()
+
+        # local counters to track inflight requests
+        self._inflight_counter = 0
+
+    def create_task(self, coro):
+        """
+        Creates a task on the engine's event loop which is running on a separate thread.
+        """
+        assert (
+            self._shutdown_event.is_set() is False
+        ), "Cannot create tasks after shutdown has been requested"
+
+        async def _wrapped_coro(coro):
+            """
+            Wraps the given coroutine in a new coroutine and decrements the in-flight counter after awaiting the original coroutine.
+            """
+            await coro
+            self._inflight_counter -= 1
+
+        # locally increment the in-flight counter
+        self._inflight_counter += 1
+        return asyncio.run_coroutine_threadsafe(_wrapped_coro(coro), self._loop)
+
+    def engine_loop(self, loop):
+        """
+        Runs the engine's event loop on a separate thread.
+        """
+        asyncio.set_event_loop(loop)
+        self._loop.run_until_complete(self.await_shutdown())
+
+    async def await_shutdown(self):
+        """
+        Primary coroutine running on the engine event loop. This coroutine is responsible for
+        keeping the engine alive until a shutdown is requested.
+        """
+        # first await the shutdown signal
+        while self._shutdown_event.is_set() is False:
+            await asyncio.sleep(5)
+
+        # then wait for all in-flight tasks to complete
+        while self._inflight_counter > 0:
+            self.logger.log_info(
+                "Awaiting remaining {} inflight requests".format(self._inflight_counter)
+            )
+            await asyncio.sleep(5)
+
+        self.logger.log_info("Shutdown complete")
+
+    def make_sample_params(self, request: CompletionRequest) -> SamplingParams:
+        """
+        Creates a vLLM SamplingParams object from the given arguments.
+        """
+        return SamplingParams(
+            n=1,
+            best_of=1,
+            temperature=request.temperature,
+            top_p=request.top_p,
+            top_k=request.top_k,
+            stop=request.stop,
+            ignore_eos=request.ignore_eos,
+            max_tokens=request.max_tokens,
+        )
+
+    def make_output_options(self, req: CompletionRequest) -> OutputOptions:
+        """
+        Creates an OutputOptions object from the given arguments.
+        """
+        return OutputOptions(
+            echo=req.echo,
+            stream=req.stream,
+        )
+
+    def preprocess(self, request) -> Tuple[str, SamplingParams, OutputOptions]:
```

Comment:
One option to simplify the pre-process, we could match the https://github.com/vllm-project/vllm/blob/0080d8329d272f06627286984be8038b0ca9d590/vllm/entrypoints/api_server.py#L21 

And use prompt, stream, sampling parameters as the input

We could always provide the prompt in the output for simplicity instead of the echo flag